README.md

```diff
@@ -129,6 +153,10 @@ python -m cdcagg_oai --help
 Note that most configuration options can be specified via command line
 arguments, configuration file options and environment variables.
 
+[Prometheus client](https://github.com/prometheus/client_python)
+provides additional configuration. Refer to its documentation for more
+information.
```

Comment:
It's unclear on what this means. Are Prometheus options not covered by `--help`?

Reply:
That's right. They are not. Prometheus Client reads its configuration options straight from environment variables using python's os.environ. It does not offer other mechanisms to change its options.

Kuha uses configargparse, which is a wrapper around Python's argparse, that provides a way to define arguments the user can use to alter options. That library also takes care of the `--help` output. I can add arbitrary text to the `--help` output paragraphs, but have no easy way to add arbitrary argument definitions, whose values would be read outside argparse.

Kuha provides a way to change the options programmatically so that the configargparse may be bypassed if needed. This is useful when using Kuha as a library / dependency and this is what is missing from Prometheus Client.

I looked at the source code of Prometheus Client. There are parts of the code that I could override to feed it options some other way, but some parts I cannot. The problematic bits are:

- [values.py::get_value_class](https://github.com/prometheus/client_python/blob/4d68ba1334f6a34ddd8117983e07ba81fb589c20/prometheus_client/values.py#L126)
- [values.py::MultiProcessValue](https://github.com/prometheus/client_python/blob/4d68ba1334f6a34ddd8117983e07ba81fb589c20/prometheus_client/values.py#L39)

If I would want to alter the behaviour, I need to override the code that is using the problematic code:

- [metrics.py::Counter._metric_init](https://github.com/prometheus/client_python/blob/4d68ba1334f6a34ddd8117983e07ba81fb589c20/prometheus_client/metrics.py#L276)
- [metrics.py::Gauge._metric_init](https://github.com/prometheus/client_python/blob/4d68ba1334f6a34ddd8117983e07ba81fb589c20/prometheus_client/metrics.py#L377)
- [metrics.py::Summary._metric_init](https://github.com/prometheus/client_python/blob/4d68ba1334f6a34ddd8117983e07ba81fb589c20/prometheus_client/metrics.py#L470)
- [metrics.py::Histogram._metric_init](https://github.com/prometheus/client_python/blob/4d68ba1334f6a34ddd8117983e07ba81fb589c20/prometheus_client/metrics.py#L582) - I'm not sure if this is needed, I have currently not used Histogram in this implementation.

So, I think it would make more sense to fork the Prometheus Client and make a few small alterations to the values.py module, than override 3-4 classes from metrics.py module, and create new implementations for values.py::get_value_class & values.py::MultiProceessValue.

Of course, if I fork the client, I would then need to maintain the fork, which means possibly more work in the future. To me, that seems a bit overkill for this small feature. Better would be to provide a PR to Prometheus Client to make its implementation more flexible.

Anyway, there are currently 2 environment variables that are used for configuration in the prometheus client: `PROMETHEUS_DISABLE_CREATED_SERIES`, and `PROMETHEUS_MULTIPROC_DIR`. We need to use the `PROMETHEUS_MULTIPROC_DIR`, if running multiple processes. These are quite clearly defined in the Prometheus Client README.md.

Reply:
Thanks. I don't think it's worth forking the Prometheus client over this.

Could you mention the environment variables in the OAI-PMH readme for completeness?

Reply:
Yes. I will mention the variables in the readme.